frontend/src/components/Edit.jsx

```diff
@@ -0,0 +1,201 @@
+    });
+  };
+
+  changeMeassurement = meassurement => {
+    this.setState({
+      currentMeassurement: meassurement
+    });
+  };
+
+  handleAdd = () => {
+    let newRecipeIngredients = this.state.recipeIngredients;
+    newRecipeIngredients.push([
+      this.state.currentIngredient,
+      this.state.currentAmount,
+      this.state.currentMeassurement
+    ]);
+
+    this.setState({
+      recipeIngredients: newRecipeIngredients
+    });
+    console.log(this.state);
+  };
+
+  handleDelete = ingredientWithAmount => {
+    let newRecipeIngredients = this.state.recipeIngredients;
+    let index = newRecipeIngredients.indexOf(ingredientWithAmount);
+    if (index !== -1) {
+      newRecipeIngredients.splice(index, 1);
+      this.setState({
+        recipeIngredents: newRecipeIngredients
+      });
+    }
+    console.log(this.state);
+  };
+
+  handleEdit = data => {
+    let ingredients = this.state.recipeIngredients;
+    if (typeof ingredients == "undefined") {
+      console.log("Ingredients cannot be empty");
+      return;
+    }
+
+    let creator = this.state.recipe.creator;
+    let id = this.state.recipe.id;
+    let recipeData = {
+      name: data.recipeName,
+      time: data.recipeTime,
+      servings: data.recipeServings,
+      ingredients: ingredients,
+      description: data.recipeDescription,
+      instructions: data.recipeInstructions,
+      creator: creator,
+      id: id
+    };
+
+    axios
+      .post("http://localhost:4000/editRecipe", recipeData)
+      .then(() => {
+        this.props.toastOpen({
+          text: "Edited!"
+        });
+      })
+      .catch(err => {
+        console.log(err);
+        this.props.toastOpen({
+          text: "Something went wrong"
+        });
+      });
+  };
+
+  render() {
+    let currentRecipe = this.state.recipe;
+
+    return (
+      <div className="editBody">
+        <div className="editTitle">
+          <DigitText.Heading3 text={"Edit"} />
+        </div>
+        <div className="editForm">
+          <DigitForm
+            onSubmit={(values, actions) => {
+              this.handleEdit(values);
+            }}
+            initialValues={{
+              recipeName: currentRecipe.name,
+              recipeTime: currentRecipe.time,
+              recipeServings: currentRecipe.servings,
+              recipeDescription: currentRecipe.description,
+              recipeInstructions: currentRecipe.instructions
+            }}
+            validationSchema={yup.object().shape({
+              recipeName: yup.string().required("This can't be empty"),
+              recipeTime: yup.string().required("This can't be empty"),
+              recipeServings: yup.string().required("This can't be empty"),
+              recipeDescription: yup.string().required("This can't be empty"),
+              recipeInstructions: yup.string().required("This can't be empty")
+            })}
+            render={({ errors }) => (
+              <DigitDesign.Card abswidth="460px">
```

Comment:
Should be `absWidth`. Also this will not work on phones that has a width of 320px (the smallest phone atm that should be supported) .